daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -157,6 +160,12 @@ getSdk damlPath projectPathM =
                     config <- readConfig path
                     fromRightM throwIO (parseVersion config)
 
+        sdkPathFallback :: DamlPath -> Maybe SdkVersion -> IO (Maybe SdkPath)
+        sdkPathFallback damlPath sdkVersionM = do
```

Comment:
I find this block pretty confusing. I'd just pattern match on sdkVersionM and simplify it a lot.

Reply:
:+1: